src/components/filter/EventsFilterPanel.tsx

```diff
@@ -60,23 +70,34 @@ function EventsFilterPanel() {
 
 	const onSubmit = React.useCallback(() => {
 		if (filter) {
+			const timestamp = Date.now();
```

Comment:
I think this logic belongs to FilterStore, and we should call it inside eventsStore.applyFilter